collection.go

```diff
@@ -23,38 +31,85 @@ type IndexingPolicy struct {
 }
 
 type IndexingMode string
-
-const (
-	Consistent = IndexingMode("Consistent")
-	Lazy       = IndexingMode("Lazy")
-)
+type OfferThroughput int32
+type OfferType string
+
+//const (
+//	Consistent = IndexingMode("Consistent")
+//	Lazy       = IndexingMode("Lazy")
+//)
+//
+//const (
+//	OfferTypeS1 = OfferType("S1")
+//	OfferTypeS2 = OfferType("S2")
+//	OfferTypeS3 = OfferType("S3")
+//)
 
 type PartitionKey struct {
 	Paths []string `json:"paths"`
 	Kind  string   `json:"kind"`
 }
 
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-collection
 type CollectionCreateOptions struct {
 	Id             string          `json:"id"`
 	IndexingPolicy *IndexingPolicy `json:"indexingPolicy,omitempty"`
 	PartitionKey   *PartitionKey   `json:"partitionKey,omitempty"`
+
+	// RTUs [400 - 250000]. Do not use in combination with OfferType
+	OfferThroughput OfferThroughput `json:"offerThroughput,omitempty"`
+	// S1,S2,S3. Do not use in combination with OfferThroughput
+	OfferType OfferType `json:"offerType,omitempty"`
 }
 
 type CollectionReplaceOptions struct {
+	Resource
 	Id             string          `json:"id"`
 	IndexingPolicy *IndexingPolicy `json:"indexingPolicy,omitempty"`
 	PartitionKey   *PartitionKey   `json:"partitionKey,omitempty"`
 }
 
+// https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-collection
 func (c *Client) CreateCollection(ctx context.Context, dbName string,
 	colOps CollectionCreateOptions, ops *RequestOptions) (*Collection, error) {
 
-	return nil, ErrorNotImplemented
+	headers := make(map[string]string)
+
+	if colOps.OfferThroughput > 0 {
+		headers[HEADER_OFFER_THROUGHPUT] = fmt.Sprintf("%d", colOps.OfferThroughput)
+	}
+
+	if colOps.OfferThroughput >= 10000 && colOps.PartitionKey == nil {
+		return nil, errors.New(fmt.Sprintf("Must specify PartitionKey for collection '%s' when OfferThroughput is >= 10000", colOps.Id))
```

Comment:
Maybe, it's not essential (up to you):
let's make this an error you can handle nicely as a caller -> either a package level variable that can be compared or, if you want to capture some context, a custom error type that can be type switched on.